game-core/src/main/java/org/triplea/game/server/HeadlessServerSetupPanelModel.java

```diff
@@ -21,7 +39,64 @@
 
   @Override
   public void showSelectType() {
-    new ServerModel(gameSelectorModel, this, null).createServerMessenger();
+    new ServerModel(gameSelectorModel, this, null, new LaunchAction() {
+      @Override
+      public void handleGameInterruption(final GameSelectorModel gameSelectorModel, final ServerModel serverModel) {
+        try {
+          log.info("Game ended, going back to waiting.");
+          // if we do not do this, we can get into an infinite loop of launching a game,
+          // then crashing out, then launching, etc.
+          serverModel.setAllPlayersToNullNodes();
+          final File f1 = getAutoSaveFileUtils().getHeadlessAutoSaveFile();
```

Comment:
With the new scoping, `f1` is probably not needed as a variable name and this could be renamed to `autoSaveFile`